tests/extmod/vfs_posix_ilistdir_del.py

```diff
@@ -0,0 +1,40 @@
+# Test ilistdir __del__ for VfsLittle using a RAM device
+import gc
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+    vfs.mkdir("/test_d1")
+    vfs.mkdir("/test_d2")
+    vfs.mkdir("/test_d3")
+
+    for i in range(10):
+        print(i)
+        idir = vfs.ilistdir("/")
+        print(any(idir))
+        for dname, *_ in vfs.ilistdir("/"):
+            vfs.rmdir(dname)
+            break
+        vfs.mkdir(dname)
+        gc.collect()
+        vfs.open("/test", "w").close()
+        vfs.remove("/test")
+
+
+tempdir = "vfs_posix_ilistdir_del.tmp"
+try:
+    os.mkdir(tempdir)
+except OSError:
+    for td in os.listdir(tempdir):
+        os.rmdir("/".join((tempdir, td)))
```

Comment:
I think if `tempdir` already exists it should just skip the test.  This would follow `vfs_posix.py`.

Reply:
sure, I guess if the folder already exists (in ci) it means the test has been run already.

Reply:
The folder should be removed when the test finishes, like `vfs_posix.py` does it.